src/components/MultiStepIndicator/MultiStepIndicator.jsx

```diff
@@ -18,40 +19,66 @@ const MultiStepIndicator = forwardRef(
       stepComponentClassName,
       dividerComponentClassName,
       fulfilledStepIcon,
-      fulfilledStepIconType
+      fulfilledStepIconType,
+      onClick,
+      textPlacement
     },
     ref
   ) => {
     const componentRef = useRef(null);
     const mergedRef = useMergeRefs({ refs: [ref, componentRef] });
     const baseClassName = "multi-step-indicator--wrapper";
     const defaultDividerClassName = `${baseClassName}__divider`;
+
+    const renderHorizontalStepIndicator = (step, index) => {
+      return (
+        <>
+          <StepIndicator
+            {...step}
+            stepNumber={index + 1}
+            type={type}
+            stepComponentClassName={stepComponentClassName}
+            fulfilledStepIcon={fulfilledStepIcon}
+            fulfilledStepIconType={fulfilledStepIconType}
+            onClick={onClick}
+          />
+          {index !== steps.length - 1 && <Divider classname={cx(defaultDividerClassName, dividerComponentClassName)} />}
+        </>
+      );
+    };
+
+    const renderVerticalStepIndicator = (step, index) => {
+      return (
+        <>
+          <VerticalStepIndicator
+            {...step}
+            stepNumber={index + 1}
+            type={type}
+            stepComponentClassName={stepComponentClassName}
+            fulfilledStepIcon={fulfilledStepIcon}
+            fulfilledStepIconType={fulfilledStepIconType}
+            onClick={onClick}
+            isFollowedByDivider={index !== steps.length - 1}
+            stepDividerClassName={cx(defaultDividerClassName, dividerComponentClassName)}
+          />
+        </>
+      );
+    };
+
+    const stepRenderer =
```

Comment:
please add useMemo on this calculation and include the two render functions within it